pkg/agent/endpoints/sds/handler.go

```diff
@@ -286,9 +288,10 @@ func (h *Handler) buildResponse(versionInfo string, req *api_v2.DiscoveryRequest
 	}
 
 	// TODO: verify the type url
+	_, defaultBundleNameBool := names[h.c.SDSDefaultBundleName]
```

Comment:
This won't work as expected if somebody passes an empty string in the `req.ResourceNames`. An empty string wouldn't matter before since it wouldn't match the bundle or certificate names. To fix this, we should not add empty resource names to the `names` map above.